chatcommands.py

```diff
@@ -1092,6 +1092,48 @@ def report(msg, urls):
         return os.linesep.join(output)
 
 
+# noinspection PyIncorrectDocstring,PyUnusedLocal
+@command(str, whole_msg=True, privileged=True, aliases=['scan', 'test-p'])
+def checkpost(msg, url):  # FIXME: Currently does not support batch report
+    """
+    Force Smokey to scan a post even if it has no recent activity
+    :param msg:
+    :param url:
+    :return I don't know:
+    """
+
+    crn, wait = can_report_now(msg.owner.id, msg._client.host)
+    if not crn:
+        raise CmdException("You should totally drop that and try jQuery.")
+
+    post_data = api_get_post(url)
+
+    if post_data is None:
+        raise CmdException("That does not look like a valid post URL")
+
+    if post_data is False:
+        raise CmdException("Cannot find data for this post in the API. "
+                           "It may have already been deleted.")
+
+    is_spam, reasons, why = check_if_spam(title=post_data.title,
```

Comment:
`check_if_spam` doesn't work this way. You need a Post (not a PostData either). The best way to handle this is to do `check_if_spam(Post(api_response=post_data.as_dict()))`

However, I'm not sure why you're fetching the post separately, scanning it, then enqueueing it to fetch and scan again. If you want to immediately report the post if it's spam, just call into `spamhandling.handle_spam`

Reply:
Very useful information. I'm doing one last modification today before going to sleep. Will change in a few minutes.